src/linter/types.go

```diff
@@ -149,6 +153,34 @@ func (conv *phpdocTypeConverter) mapShapeType(params []phpdoc.TypeExpr) []meta.T
 	return []meta.Type{{Elem: shape.name}}
 }
 
+func (conv *phpdocTypeConverter) mapTupleType(params []phpdoc.TypeExpr) []meta.Type {
+	types := make([]phpdoc.TypeExpr, 0, len(params))
+	for i, p := range params {
+		if p.Value == "*" || p.Value == "..." {
+			continue
+		}
+		if p.Kind != phpdoc.ExprName && p.Kind != phpdoc.ExprGeneric {
```

Comment:
Do we need this kind of restriction?

I think we can remove this check and leave all validation to `mapShapeType`.